vms/evm/subscriber.go

```diff
@@ -61,6 +61,7 @@ type subscriber struct {
 
 	logger logging.Logger
 	db     database.RelayerDatabase
+	dial   func(url string) (ethclient.Client, error)
```

Comment:
What's the motivation for adding this as a field? It doesn't look like we're adding any abstraction on top of `ethclient.Client`.

Reply:
Oh I see, we're using it to mock `ethclient.Dial` in the unit test. Can you add a comment explaining this?

Reply:
added a comment in https://github.com/ava-labs/awm-relayer/pull/89/commits/6618b41f81d0bc8edfda6b83c1f21b9a7856ceca